x-pack/plugins/ml/public/application/services/elastic_models_service.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { type ModelDefinitionResponse } from '@kbn/ml-trained-models-utils';
+import { type TrainedModelsApiService } from './ml_api_service/trained_models';
+
+export type ElserVersion = 1 | 2;
+
+export class ElasticModels {
+  constructor(private readonly trainedModels: TrainedModelsApiService) {}
+
+  /**
+   * Provides an ELSER model name and configuration for download based on the current cluster architecture.
+   * The current default version is 2. If running on Cloud it returns the Linux x86_64 optimized version.
+   * If any of the ML nodes run a different OS rather than Linux, and the CPU architecture isn't x86_64,
+   * a portable version of the model is returned.
+   */
+  public async getELSER(options?: { version?: ElserVersion }): Promise<ModelDefinitionResponse> {
```

Comment:
We may need a server side version of this function, shared out with our other shared functions. I guess it depends on how other plugins want to consume it.

Reply:
Refactored in https://github.com/elastic/kibana/pull/167000/commits/d97cf1db330af7814a4d8a3895e1b3c918ba63fa